src/Core/Mesh/TriangleMesh.hpp

```diff
@@ -26,9 +26,14 @@ namespace Ra
             /// Appends another mesh to this one.
             inline void append( const TriangleMesh& other );
 
+            /// Triangulates quads
+            inline void triangulate();
+
             VectorArray<Vector3>  m_vertices;
             VectorArray<Vector3>  m_normals;
             VectorArray<Triangle> m_triangles;
+
+            VectorArray<VectorNui>  m_faces;

```

Comment:
I still believe there should be two separate classes : Triangle Mesh and Polygon Mesh